nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -133,9 +138,15 @@ protected void notifyEvent(CoreSession session, String eventType, DocumentModel
     }
 
     protected NuxeoPrincipal getAuthor(DocumentModel docModel) {
-        String[] contributors = (String[]) docModel.getProperty("dublincore", "contributors");
-        UserManager userManager = Framework.getService(UserManager.class);
-        return userManager.getPrincipal(contributors[0]);
+        String author = null;
+        if (docModel.hasSchema(COMMENT_SCHEMA)) {
+            // means annotation / comment
+            author = (String) docModel.getPropertyValue(COMMENT_AUTHOR);
+        } else if (StringUtils.isEmpty(author)) {
+            String[] contributors = (String[]) docModel.getProperty("dublincore", "contributors");
```

Comment:
i don't know if we can find ourself with a comment without author, because if it is the case and the docModel is a Comment, you will not evaluate the second if and we find ourself without author; why if we change to:

```
         if (docModel.hasSchema(COMMENT_SCHEMA)) {
            // means annotation / comment
            author = (String) docModel.getPropertyValue(COMMENT_AUTHOR);
        }
        if (StringUtils.isEmpty(author)) {
```

Reply:
Yes, changed it for a simple if